src/components/sub-components/mgt-flyout/mgt-flyout.ts

```diff
@@ -76,137 +132,168 @@ export class MgtFlyout extends LitElement {
-
-        // Rules of alignment:
-        // 1. Center if possible
-        // 2. Don't cross screen boundaries.
-
-        if (flyoutWidth > windowRect.width) {
-          // page width is smaller than flyout, render all the way to the left
-          left = -anchorRect.left;
-        } else if (Math.floor(anchorRect.width) >= flyoutWidth) {
-          // anchor is larger than flyout, render aligned to anchor
-          left = anchorRect.left;
-        } else {
-          const centerOffset = Math.floor(flyoutWidth / 2 - anchorRect.width / 2);
-
-          if (anchorRect.left - centerOffset < windowRect.left) {
-            // centered flyout is off screen to the left, render on the left edge
-            left = windowRect.left - anchorRect.left;
-          } else if (flyoutRight - centerOffset > windowRect.width) {
-            // centered flyout is off screen to the right, render on the right edge
-            left =
-              -centerOffset * 2 -
-              (anchorRect.left +
-                Math.floor(anchorRect.width) -
-                Math.min(windowRect.left + windowRect.width, innerWidth)) -
-              1;
-          } else {
-            // render centered
-            left = -centerOffset;
+      const anchorRect = anchor.getBoundingClientRect();
+      const anchorCenterX = anchorRect.left + anchorRect.width / 2;
+      const anchorCenterY = anchorRect.top + anchorRect.height / 2;
+
+      const windowRect: IWindowSegment = {
+        height: 0,
+        left: 0,
+        top: 0,
+        width: 0
+      };
+
+      if (isWindowSegmentAware()) {
+        const segmentAwareWindow = getSegmentAwareWindow();
+        const screenSegments = segmentAwareWindow.getWindowSegments();
+
+        let anchorSegment: IWindowSegment;
+        for (const segment of screenSegments) {
+          if (anchorCenterX >= segment.left && anchorCenterY >= segment.top) {
+            anchorSegment = segment;
           }
         }
 
-        if (flyoutHeight > windowRect.height || (windowRect.height < flyoutBottom && anchorRect.top < flyoutHeight)) {
-          top = -flyoutTop + anchorRect.height;
-        } else if (windowRect.height < flyoutBottom) {
-          bottom = anchorRect.height;
+        windowRect.left = anchorSegment.left;
+        windowRect.top = anchorSegment.top;
+        windowRect.width = anchorSegment.width;
+        windowRect.height = anchorSegment.height;
+      } else {
+        windowRect.height = innerHeight;
+        windowRect.width = innerWidth;
+      }
+
+      // normalize flyoutrect since we could have moved it before
+      // need to know where would it render, not where it renders
+      const flyoutWidth = flyout.scrollWidth;
+      const flyoutHeight = flyout.scrollHeight;
+      const flyoutTop = anchorRect.bottom;
+      const flyoutLeft = anchorRect.left;
+      const flyoutRight = flyoutLeft + flyoutWidth;
+      const flyoutBottom = flyoutTop + flyoutHeight;
+
+      // Rules of alignment:
```

Comment:
I'm not sure centering is a good idea especially for the login component and the people picker component. We should stick with aligning with the anchor

Reply:
I didn't actually adjust any of this logic. I just moved the `isOpen` check to the top. I saw you had some improved positioning logic in your flyout PR though. We should switch to use yours in a follow up PR :)